### ISSUE TITLE
Tensorflow Lite not support Siamese network with parameter sharing

### ISSUE THREAD

**Issue by guohengkai** [tensorflow/tensorflow#17984]
Hi,

I found that the Tensorflow Lite did not support Siamese network with parameter sharing of conv2d layers and batch normalization layers yet. (e.g. EnsureBiasVectors will fail to deal with different names of conv output layer with the same kernel.) Is there any plan to support it?

Thanks.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by guohengkai** [comment]
Hi, sorry for that. And I found the problems were caused by the transformation (e.g. removing of original kernel of convolution after transformation) of operators. 

Have I written custom code
No

OS Platform and Distribution
Mac OS

TensorFlow installed from
Compilation from github

TensorFlow version
r1.6

Bazel version
N/A

CUDA/cuDNN version
CPU only

GPU model and memory
N/A

Exact command to reproduce
bazel-bin/tensorflow/contrib/lite/toco/toco

**Comment by aselle** [comment]
@guohengkai, it's difficult to help you without a reproducible test case. Could you provide one?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by guohengkai** [comment]
@aselle Sorry for the late response. The following codes will save the model into "test.pbtxt", which will cause this problem.
`import tensorflow as tf

with tf.Graph().as_default() as graph:
    # Input
    input_placeholder = tf.placeholder(tf.float32, shape=(1, 20, 20, 2))
    input_split_1, input_split_2 = tf.split(input_placeholder, 2, axis=3)

    # Stream 1
    conv1 = tf.layers.conv2d(input_split_1, 8, kernel_size=3, strides=1,
                            padding='SAME', activation=None, name='conv1')
    bn1 = tf.layers.batch_normalization(conv1, axis=3, momentum=0.997,
                                        epsilon=1e-5, center=True, scale=True,
                                        training=True, fused=True, name='bn1')
    relu1 = tf.nn.relu(bn1)
    conv2 = tf.layers.conv2d(relu1, 8, kernel_size=3, strides=1,
                            padding='SAME', activation=None, name='conv2')
    bn2 = tf.layers.batch_normalization(conv2, axis=3, momentum=0.997,
                                        epsilon=1e-5, center=True, scale=True,
                                        training=True, fused=True, name='bn2')
    relu2 = tf.nn.relu(bn2)

    # Stream 2
    conv1_2 = tf.layers.conv2d(input_split_2, 8, kernel_size=3, strides=1,
                            padding='SAME', activation=None, name='conv1',
                            reuse=True)
    bn1_2 = tf.layers.batch_normalization(conv1_2, axis=3, momentum=0.997,
                                        epsilon=1e-5, center=True, scale=True,
                                        training=True, fused=True, name='bn1',
                                        reuse=True)
    relu1_2 = tf.nn.relu(bn1_2)
    conv2_2 = tf.layers.conv2d(relu1_2, 8, kernel_size=3, strides=1,
                            padding='SAME', activation=None, name='conv2',
                            reuse=True)
    bn2_2 = tf.layers.batch_normalization(conv2_2, axis=3, momentum=0.997,
                                        epsilon=1e-5, center=True, scale=True,
                                        training=True, fused=True, name='bn2',
                                        reuse=True)
    relu2_2 = tf.nn.relu(bn2_2)

    # Final
    concat = tf.concat(axis=3, values=[relu2, relu2_2])
    flatten = tf.layers.flatten(concat)
    fc = tf.layers.dense(flatten, 200)

    tf.train.write_graph(graph.as_graph_def(), '.', 'test.pbtxt')
`

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrehentz** [comment]
It looks like you will have to freeze your graph. Does https://www.tensorflow.org/extend/tool_developers/#freezing help?

**Comment by guohengkai** [comment]
@andrehentz Thanks for your kind response. Indeed I have frozen my graph. The codes I provided are just for network structure.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by anitha-v** [comment]
We will try this, can you please share the error message if you have it handy ?

**Comment by guohengkai** [comment]
@anitha-v Thanks for your response. I do not have any error message handy now. Sorry for that. The errors are something like dimension inconsistent with conv due to the trimming operation. But I think it's easy to reproduce.

**Comment by aselle** [comment]
I'll take a look when I can. You may be able to work around this, by creating a new network that duplicates the weights.

**Comment by guohengkai** [comment]
@aselle OK. Thanks.

**Comment by raghuraman-k** [comment]
Have you created an eval graph? Your example seems to use batch normalization with training. Can you try with Training=False?

**Comment by guohengkai** [comment]
@raghuraman-k Hi, the provided example is just a demo to show the network structure I used because my codes are more complicated. I have converted it into an eval graph.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 33 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 49 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 64 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 79 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 94 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aselle: It has been 109 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by andrehentz** [comment]
Hi there! I'll close this issue for now. Please open a separate issue if you are still encountering this problem with the latest code.

**Comment by JJKK1313** [comment]
still not working though.
cannt work on siamese network with tflite
